### PR TITLE
Introduced MIGRATION_ASSESSMENT_STATS table in metaDB for storing metric about table sizes, iops, columns info, indexes etc

### PR DESCRIPTION
**Pull request by sanyamsinghal**
- also converted and stored other CSV files into tables - TABLE_INDEX_IOPS, TABLE_INDEX_SIZES, TABLE_ROW_COUNTS, COLUMNS_COUNT, INDEX_TO_TABLE_MAPPING, TABLE_COLUMNS_DATA_TYPES

### CODE REVIEW

File: yb-voyager/src/migassessment/assessmentDB.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	err = file.Close()
+	if err != nil {
+		return fmt.Errorf("error closing assessment db file %s: %w", assessmentDBPath, err)
+	}
+	return nil
+}
+
+func initAssessmentDB(assessmentDBPath string) error {
+	log.Infof("initializing assessment db at %s", assessmentDBPath)
+	conn, err := sql.Open("sqlite3", fmt.Sprintf("%s%s", assessmentDBPath, metadb.SQLITE_OPTIONS))
```
Comment (amit-yb) [inline yb-voyager/src/migassessment/assessmentDB.go:75]: Doesn't this create a new file if it doesn't exist? If it does, do we need `createAssessmentDBFile()`?

File: yb-voyager/src/migassessment/assessmentDB.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+	return &AssessmentDB{db: db}, nil
+}
+
+func (adb *AssessmentDB) BulkInsert(table string, records [][]string) error {
+	tx, err := adb.db.Begin()
+	if err != nil {
+		return fmt.Errorf("error starting transaction for bulk insert into %s: %w", table, err)
+	}
+
```
Comment (amit-yb) [inline yb-voyager/src/migassessment/assessmentDB.go:162]: `defer tx.Rollback(ctx)` ???

File: yb-voyager/src/migassessment/assessmentDB.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	columnNames := records[0]
+	stmtStr := fmt.Sprintf(`INSERT INTO %s (%s) VALUES (%s)`, table,
```
Comment (amit-yb) [inline yb-voyager/src/migassessment/assessmentDB.go:164]: `strings.Join(strings.Split(strings.Repeat("?", 5), ""), ", ")`

File: yb-voyager/src/migassessment/assessmentDB.go [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+	_, err := adb.db.Exec(INSERT_TABLE_STATS)
+	if err != nil {
+		return fmt.Errorf("error executing INSERT_TABLE_STATS on %s table: %w", MIGRATION_ASSESSMENT_STATS, err)
+	}
+
+	_, err = adb.db.Exec(INSERT_INDEX_STATS)
+	if err != nil {
+		return fmt.Errorf("error executing INSERT_INDEX_STATS on %s table: %w", MIGRATION_ASSESSMENT_STATS, err)
+	}
```
Comment (amit-yb) [inline yb-voyager/src/migassessment/assessmentDB.go:226]: Make an array of statements and Exec them in a loop.

File: yb-voyager/src/migassessment/assessmentDB.go [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
```
Comment (amit-yb) [inline yb-voyager/src/migassessment/assessmentDB.go:227]: Who is going to populate these tables?

File: yb-voyager/src/metadb/metadataDB.go [hunk 6]
```diff
@@ -610,3 +610,19 @@ func (m *MetaDB) ResetQueueSegmentMeta(importerRole string) error {
...
+	return sqlStmt, nil
+}
+
+func (m *MetaDB) ExecStatement(stmt string) error {
+	_, err := m.db.Exec(stmt)
+	if err != nil {
+		return fmt.Errorf("error while executing statement on metaDB: %w", err)
+	}
+	return nil
+}
```
Comment (amit-yb) [inline yb-voyager/src/metadb/metadataDB.go:628]: Are they used anywhere? If not, drop em.

**Review by amit-yb — APPROVED** [review]
LGTM with a few minor comments.

File: yb-voyager/src/srcdb/data/gather-assessment-data/postgresql/table-columns-data-types.sql
[the remark this answers is not in the document]
Comment (sanyamsinghal) [reply yb-voyager/src/srcdb/data/gather-assessment-data/postgresql/table-columns-data-types.sql:1]: some issue, even after renaming both of these are `.psql` and `.sql` for table-columns-data-types.
Probably some merge conflict resolution caused this.

File: yb-voyager/src/migassessment/assessmentDB.go [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createAssessmentDBFile(assessmentDBPath string) error {
+	log.Infof("creating assessment db file at %s", assessmentDBPath)
+	file, err := os.Create(assessmentDBPath)
```
[the remark this answers is not in the document]
Comment (sanyamsinghal) [reply yb-voyager/src/migassessment/assessmentDB.go]: removed this file creation code.

File: yb-voyager/src/migassessment/assessmentDB.go (hunk 4 above)
Comment (sanyamsinghal) [reply yb-voyager/src/migassessment/assessmentDB.go]: done

Comment (sanyamsinghal) [reply yb-voyager/src/migassessment/assessmentDB.go]: dne

**Pull request merged**